sdks/java/core/src/test/java/org/apache/beam/sdk/runners/dataflow/TestCountingSource.java

```diff
@@ -164,21 +168,20 @@ public CountingSourceReader(int startingPoint) {
 
     @Override
     public boolean start() {
-      return true;
+      return advance();
```

Comment:
Contractually you can only be called to start once, use another variable to track whether you have been started and throw an error if you have been called multiple times.

Also, check that in advance, that start has been called.